python/pyarrow/tests/test_compute.py

```diff
@@ -1731,6 +1731,22 @@ def test_cast():
     assert pc.cast(arr, expected.type) == expected
 
 
+def test_fsl_to_fsl_cast():
+    for value_type in (pa.bool_(), pa.float32(), pa.int64()):
+        # Valid case
+        # Different field name and different type.
+        cast_type = pa.list_(pa.field("element", value_type), 2)
+        fsl = pa.FixedSizeListArray.from_arrays(
+            pa.array([1, 2, 3, 4, 5, 6]), 2)
```

Comment:
specify the type of the source array explicitly here?

Reply:
It would be good to test with nulls as well, both at the list level and the child array level

Reply:
I believe that I have addressed the points. (1st and 2nd in Python test and 3rd in C++ test)

Sorry, I am new to the project so still learning the jargon (and might have missed to address the point) :)